lib/index.js

```diff
@@ -347,9 +347,9 @@ Server.prototype.of = function(name, fn){
  */
 
 Server.prototype.close = function(){
-  this.nsps['/'].sockets.forEach(function(socket){
-    socket.onclose();
-  });
+  for (var id in this.nsps['/'].sockets) {
```

Comment:
You're not checking for `hasOwnProperty` here. Extensions fo the natives will make socket.io break here. Test needed for that scenario as well.